system/system.proto

```diff
@@ -149,11 +152,21 @@ message RebootStatusRequest {
 }
 
 message RebootStatusResponse {
-  bool active = 1;      // If reboot is active.
-  uint64 wait = 2;      // Time left until reboot.
-  uint64 when = 3;      // Time to reboot in nanoseconds since the epoch.
-  string reason = 4;    // Reason for reboot.
-  uint32 count = 5;     // Number of reboots since active.
+  bool active = 1 [deprecated = true];  // If reboot is active.
+  uint64 wait = 2 [deprecated = true];  // Time left until reboot.
+  uint64 when = 3
+      [deprecated = true];  // Time to reboot in nanoseconds since the epoch.
+  string reason = 4 [deprecated = true];  // Reason for reboot.
+  uint32 count = 5 [deprecated = true];   // Number of reboots since active.
+
+  message Status {
+    bool active = 1;    // If reboot is active.
+    uint64 wait = 2;    // Time left until reboot.
+    uint64 when = 3;    // Time to reboot in nanoseconds since the epoch.
+    string reason = 4;  // Reason for reboot.
+    uint32 count = 5;   // Number of reboots since active.
+  }
+  repeated Status statuses = 6;
```

Comment:
#not sure the logics correctly - if send a statusrequest that is empty what is the expected return? a statuses with a len of 1? 

what if there is no reboot in process?
does the system return a single status message saying no reboot or empty statuses

Reply:
should the Status have the path that is been queried as part of the response?

Reply:
* ah, good catch. I copied and pasted the fields too fast and forgot to add the subcomponent paths. I just added it. 

* Reading the existing fields (e.g. `count, active`), seems like the original idea is to return reboot status (i.e. active=false) even no active reboot or planned reboot on the sub-component? The original idea makes sense to me. `RebootStatusRequest` reports all reboots activities which covers the history and planning. Therefore, to align with the original idea I think we can clarify here saying that if sub-component not set, return status on all sub-components. Thoughts?

Reply:
@xw-g The last entry of Reboot Status is covered in the openconfig-platform: last-reboot-reason.

```proto
    leaf last-reboot-reason {
      type identityref {
        base oc-platform-types:COMPONENT_REBOOT_REASON;
      }
      description
        "This reports the reason of the last reboot of the component.";
    }
```
Should the RebootStatusRequest be applicable to only current/planned reboot requests?

Reply:
i would return error here if unset - as the total number of components may be large